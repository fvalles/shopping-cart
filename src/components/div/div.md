Basic div component with default width='100%':

```jsx
<Div />
```

Div component with custom display, width and align-items properties set:

```jsx
<Div alignItems="flex-start" display="flex" width="330px" />
```

Div component with default width='100%' and custom display and justify-content properties set:

```jsx
<Div display="flex" justifyContent="flex-end" />
```
