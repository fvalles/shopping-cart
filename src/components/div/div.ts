import styled, { css } from 'styled-components';

/**
 * Types
 */

interface DivProps {
  /** Div align items property. Default is set to 'center' */
  alignItems?: string;
  /** Div display property */
  display?: string;
  /** Div margin */
  margin?: string;
  /** Div width. Default value is 100% */
  width?: string;
  /** Div justify-content property */
  justifyContent?: string;
}

/**
 * Div
 */

export const Div = styled.div<DivProps>`
  ${({ alignItems }) =>
    alignItems &&
    css`
      align-items: ${alignItems};
    `}
  ${({ display }) =>
    display &&
    css`
      display: ${display};
    `}
  ${({ justifyContent }) =>
    justifyContent &&
    css`
      justify-content: ${justifyContent};
    `}
  ${({ margin }) =>
    margin &&
    css`
      margin: ${margin};
    `}
  width: ${({ width = '100%' }) => width};
`;
