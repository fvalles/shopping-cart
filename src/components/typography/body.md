Basic body text component with default color="body":

```jsx
<Body>Lorem ipsum</Body>
```

Body text component with color="primary":

```jsx
<Body color="primary">Lorem ipsum</Body>
```
