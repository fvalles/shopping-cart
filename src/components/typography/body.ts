import styled from 'styled-components';
import { TypographyProps } from './common';

export const Body = styled.p<TypographyProps>`
  color: ${({ color = 'body', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 13px;
  font-weight: 400;
  letter-spacing: -0.01em;
  line-height: 18px;
`;
