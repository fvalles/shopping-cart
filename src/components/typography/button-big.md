Basic button big text component with default color="primary":

```jsx
<ButtonBig>Lorem ipsum</ButtonBig>
```

Button big text component with color="summaryText":

```jsx
<ButtonBig color="summaryText">Lorem ipsum</ButtonBig>
```
