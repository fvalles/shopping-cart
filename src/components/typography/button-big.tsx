import styled from 'styled-components';
import { TypographyProps } from './common';

export const ButtonBig = styled.span<TypographyProps>`
  color: ${({ color = 'primary', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 25px;
`;
