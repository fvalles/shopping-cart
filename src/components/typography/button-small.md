Basic button small text component with default color="white":

```jsx
<ButtonSmall>Lorem ipsum</ButtonSmall>
```

Button small text component with color="summaryText":

```jsx
<ButtonSmall color="summaryText">Lorem ipsum</ButtonSmall>
```
