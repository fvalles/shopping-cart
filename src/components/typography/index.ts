export { ButtonBig } from './button-big';
export { ButtonSmall } from './button-small';
export { H1 } from './h1';
export { H2 } from './h2';
export { H3 } from './h3';
export { ListItemBig } from './list-item-big';
export { ListItemMedium } from './list-item-medium';
export { ListItemSmall } from './list-item-small';
export { Span } from './span';
export { Subtitle } from './subtitle';
export { Title } from './title';
