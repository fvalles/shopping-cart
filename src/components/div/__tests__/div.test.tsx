import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { H1 } from '@components/typography';
import { Div } from '../div';

describe('div component', () => {
  it('should render with default css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Div>
            <H1>Some text</H1>
          </Div>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('width', '100%');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Div
            alignItems="center"
            display="flex"
            justifyContent="center"
            margin="16px">
            <H1>Some text</H1>
          </Div>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('align-items', 'center');
    expect(result).toHaveStyleRule('display', 'flex');
    expect(result).toHaveStyleRule('justify-content', 'center');
    expect(result).toHaveStyleRule('margin', '16px');
    expect(result).toHaveStyleRule('width', '100%');
    expect(result).toMatchSnapshot();
  });
});
