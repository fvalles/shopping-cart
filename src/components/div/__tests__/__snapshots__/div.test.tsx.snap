// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`div component should render with custom css properties 1`] = `
.c1 {
  color: #212240;
  font-family: Avenir Medium,'Avenir Next','Segoe UI', sans-serif;
  font-size: 18px;
  -webkit-letter-spacing: -0.18px;
  -moz-letter-spacing: -0.18px;
  -ms-letter-spacing: -0.18px;
  letter-spacing: -0.18px;
  line-height: 24px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 16px;
  width: 100%;
}

<div
  className="c0"
  display="flex"
>
  <h1
    className="c1"
  >
    Some text
  </h1>
</div>
`;

exports[`div component should render with default css properties 1`] = `
.c1 {
  color: #212240;
  font-family: Avenir Medium,'Avenir Next','Segoe UI', sans-serif;
  font-size: 18px;
  -webkit-letter-spacing: -0.18px;
  -moz-letter-spacing: -0.18px;
  -ms-letter-spacing: -0.18px;
  letter-spacing: -0.18px;
  line-height: 24px;
}

.c0 {
  width: 100%;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Some text
  </h1>
</div>
`;
